client/beefy/src/worker.rs

```diff
@@ -482,13 +502,42 @@ where
 			)?,
 			RoundAction::Enqueue => {
 				debug!(target: "beefy", "🥩 Buffer vote for round: {:?}.", block_num);
-				self.pending_votes.entry(block_num).or_default().push(vote)
+				self.enqueue_votes(block_num, vote);
```

Comment:
You can also use the original `BTreeMap` and just add a check inline right here:

```suggestion
				if self.pending_votes.len() < MAX_BUFFERED_VOTE_ROUNDS {
					let votes_vec = self.pending_votes.entry(block_num).or_default();
					if votes_vec.try_push(vote).is_err() {
						warn!(target: "beefy", "🥩 Buffer vote dropped for round: {:?}", block_num)
					}
				} else {
					error!(target: "beefy", "🥩 Buffer justification dropped for round: {:?}.", block_num);
				}
```

This is the single place where the votes map is being inserted into.

And you don't need the big `fn enqueue_votes()` either.